Notebook/Reactjs/learning-hooks/src/useWindowDimensions.js

```diff
@@ -1,31 +1,28 @@
 import { useState, useEffect } from "react";
 
+function debounce(callback, wait) {
+  let timeout;
+  return (...args) => {
+    const context = this;
+    clearTimeout(timeout);
+    timeout = setTimeout(() => callback.apply(context, args), wait);
+  };
+}
+
```

Comment:
This function doesn't depend on the component's state, so it makes sense to move it out. This is also more performant.